cmd/agent/app/agent_test.go

```diff
@@ -88,6 +88,6 @@ func TestAgentStartStop(t *testing.T) {
 	body, err := ioutil.ReadAll(resp.Body)
 	assert.NoError(t, err)
 	assert.Equal(t, "tcollector error: no peers available\n", string(body))
-	agent.Stop()
 	assert.NoError(t, <-ch)
```

Comment:
isn't this going to deadlock? Nothing writing to this channel until we call Stop